src/main/java/net/sf/jabref/logic/importer/fileformat/ModsImporter.java

```diff
@@ -0,0 +1,427 @@
+        }
+    }
+
+    private void parseRecordInfo(Map<String, String> fields, RecordInfoDefinition recordInfo) {
+        List<JAXBElement<?>> recordContent = recordInfo.getRecordContentSourceOrRecordCreationDateOrRecordChangeDate();
+        for (JAXBElement<?> jaxbElement : recordContent) {
+            Object value = jaxbElement.getValue();
+            if (value instanceof StringPlusLanguagePlusAuthority) {
+                StringPlusLanguagePlusAuthority source = (StringPlusLanguagePlusAuthority) value;
+                putIfValueNotNull(fields, "source", source.getValue());
+            } else if (value instanceof LanguageDefinition) {
+                LanguageDefinition language = (LanguageDefinition) value;
+                List<String> languages = new ArrayList<>();
+                List<LanguageTermDefinition> languageTerms = language.getLanguageTerm();
+                for (LanguageTermDefinition languageTerm : languageTerms) {
+                    languages.add(languageTerm.getValue());
+                }
+                putIfListIsNotEmpty(fields, languages, FieldName.LANGUAGE, ", ");
+            }
+        }
+    }
+
+    private void parseRelatedModsGroup(Map<String, String> fields, List<Object> relatedModsGroup) {
+        for (Object groupElement : relatedModsGroup) {
+            if (groupElement instanceof PartDefinition) {
+                PartDefinition part = (PartDefinition) groupElement;
+                List<Object> detailOrExtentOrDate = part.getDetailOrExtentOrDate();
+                for (Object object : detailOrExtentOrDate) {
+                    if (object instanceof DetailDefinition) {
+                        DetailDefinition detail = (DetailDefinition) object;
+                        List<JAXBElement<StringPlusLanguage>> numberOrCaptionOrTitle = detail
+                                .getNumberOrCaptionOrTitle();
+
+                        //In the for loop should only be the value of the element that belongs to the detail not be null
+                        for (JAXBElement<StringPlusLanguage> jaxbElement : numberOrCaptionOrTitle) {
+                            StringPlusLanguage value = jaxbElement.getValue();
+                            putIfValueNotNull(fields, detail.getType(), value.getValue());
+                        }
+                    } else if (object instanceof ExtentDefinition) {
+                        ExtentDefinition extentDefinition = (ExtentDefinition) object;
+                        putPageInformation(extentDefinition, fields);
+                    }
+                }
+            } else if (groupElement instanceof TitleInfoDefinition) {
+                TitleInfoDefinition titleInfo = (TitleInfoDefinition) groupElement;
+                List<Object> titleOrSubTitleOrPartNumber = titleInfo.getTitleOrSubTitleOrPartNumber();
+                for (Object object : titleOrSubTitleOrPartNumber) {
+                    if (object instanceof JAXBElement) {
```

Comment:
Maybe you should looked at an old commit. Now there are two if's and I think using a stream here would make it much more unreadable.